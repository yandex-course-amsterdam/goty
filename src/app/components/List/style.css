.list {
  display: flex;
  flex-direction: column;

  list-style-type: none;

  padding: 10px 0 0;
  margin: 0;
}
