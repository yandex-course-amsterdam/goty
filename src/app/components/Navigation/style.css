.navigation {
  display: flex;
  flex-direction: column;
}

.title {
  font: normal bold 14px/16px 'Inter Bold', sans-serif;
  letter-spacing: 0.75px;
  color: #605b6a;
  padding: 0 0 19px 15px;
  border-bottom: 1px solid #605b6a;
}

@media (max-width: 1399px) {
  .title {
    display: none;
  }
}
