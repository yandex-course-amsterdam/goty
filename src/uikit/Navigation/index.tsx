export { Container } from './Container'
export { ListItem } from './Item'
export { List } from './List'
