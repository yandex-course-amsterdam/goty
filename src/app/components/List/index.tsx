import React, { ReactElement } from 'react'

import cn from 'classnames'

import style from './style.css'

type ListProps = {
  children?: ReactElement[] | ReactElement
  className?: string
}

export const List = ({ children, className }: ListProps): ReactElement => {
  return <ul className={cn(style.list, className)}>{children}</ul>
}
