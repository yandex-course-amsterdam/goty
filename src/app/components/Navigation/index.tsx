import React, { ReactElement } from 'react'

import style from './style.css'

type NavigationProps = {
  children?: React.ReactNode
  title?: string
}

export const Navigation = ({
  children,
  title
}: NavigationProps): ReactElement => {
  return (
    <nav className={style.navigation}>
      {title && <p className={style.title}>{title}</p>}
      {children}
    </nav>
  )
}
