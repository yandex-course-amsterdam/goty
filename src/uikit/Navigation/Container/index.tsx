import React, { FC } from 'react'

import style from './style.css'

interface IProps {
  children?: React.ReactNode
  title?: string
}

export const Container: FC<IProps> = ({ children, title }): JSX.Element => (
  <nav className={style.navigation}>
    {title && <p className={style.title}>{title}</p>}
    {children}
  </nav>
)
