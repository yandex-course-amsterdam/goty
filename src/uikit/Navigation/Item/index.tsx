import React, { FC } from 'react'
import { useRouteMatch } from 'react-router-dom'
import cn from 'classnames'

import style from './style.css'

interface IProps {
  text: string
  root?: string
  active?: boolean
  className?: string
  children?: React.ReactNode
}

export const ListItem: FC<IProps> = ({
  text,
  className,
  children,
  root
}): JSX.Element => {
  const { url } = useRouteMatch()

  return (
    <li className={cn(style.item, className, root === url && style.active)}>
      {children}
      <p className={style.text}>{text}</p>
    </li>
  )
}
