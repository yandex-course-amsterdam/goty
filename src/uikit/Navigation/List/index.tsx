import React, { FC } from 'react'

import cn from 'classnames'

import style from './style.css'

interface IProps {
  children?: React.ReactNode
  className?: string
}

export const List: FC<IProps> = ({ children, className }): JSX.Element => (
  <ul className={cn(style.list, className)}>{children}</ul>
)
