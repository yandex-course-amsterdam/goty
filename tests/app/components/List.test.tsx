import React from 'react'
import * as renderer from 'react-test-renderer'

import { List } from 'uikit/Navigation/List'

describe('List component', () => {
  test('Render itself correctly', () => {
    const component = renderer.create(<List />).toJSON()
    expect(component).toMatchSnapshot()
  })

  test('Render children correctly', () => {
    const component = renderer.create(<List>1</List>).toJSON()
    expect(component).toMatchSnapshot()
  })

  test('Use className prop', () => {
    const component = renderer.create(<List className="someClass" />).toJSON()
    expect(component).toMatchSnapshot()
  })
})
