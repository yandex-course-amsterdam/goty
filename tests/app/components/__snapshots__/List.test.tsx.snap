// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component Render children correctly 1`] = `
<ul
  className=""
>
  1
</ul>
`;

exports[`List component Render itself correctly 1`] = `
<ul
  className=""
/>
`;

exports[`List component Use className prop 1`] = `
<ul
  className="someClass"
/>
`;
